dashboard/app/helpers/users_helper.rb

```diff
@@ -214,7 +214,8 @@ def user_summary(user)
           # for now, we don't allow authorized teachers to be "locked"
           if locked && !user.authorized_teacher?
             progress[level_id] = {
-              status: LEVEL_STATUS.locked
+              status: LEVEL_STATUS.locked,
```

Comment:
Between the block at 184 and here, could you describe the relationship between status == LEVEL_STATUS.locked and the locked field?  Will they always be in agreement?  If not, when do they diverge?  How is each used by the front-end?

Reply:
LEVEL_STATUS is used for the bubble display, and can be a whole host of things (including locked, submitted, not tried, review rejected, etc). The issue here is that it was being used by stageLocked(levels) in progressHelper to determine whether or not the assessment could be accessed (what we have been referring to as lock status). This is not only undesirable, but also incorrect,  because we now allow for a case in which an assessment is submitted but unlocked, which cannot be represented with this field alone. This update makes stageLocked() check the actual source of truth, show_as_locked in user_level. 

To answer your question, these variables will be the same (locked) in only one case. After the assessment has been unlocked, the LEVEL_STATUS can change to a whole bunch of other things representing the state of the assessment completion. I think it would be an excellent move for future work to refactor this field.

Reply:
After your change, does any front-end code still key off of LEVEL_STATUS.locked?  Or does front-end code now only depend on the new locked field?

Reply:
Excellent question. It looks like this was the only place LEVEL_STATUS.locked was checked... I'll verify.

Reply:
If that's the case, we can consider removing LEVEL_STATUS.locked, either as part of this commit or as a future cleanup task that someone else can do.  :-)

Reply:
Yes- On board for this. To avoid running into potential bugs with LEVEL_STATUS in this PR, I've documented it as part of this cleanup ticket: https://codedotorg.atlassian.net/browse/LP-1857.